### PR TITLE
Documents API

### PR DESCRIPTION
**Pull request by gmourier**
This specification describes the documents API endpoints permitting to list, fetch, add/replace, and delete index documents.

## Misc
This is a necessary specification to start work on v0.28.

### CODE REVIEW

**Comment by gmourier** [comment]
test

File: text/0124-documents-api.md [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+    "overview": "William has failed to kill himself so many times that he outsources his suicide to aging assassin Leslie. But with the contract signed and death assured within a week (or his money back), William suddenly discovers reasons to live... However Leslie is under pressure from his boss to make sure the contract is completed."
+  }
+]
+```
+
+##### 3.1.1.4. Errors
+
+- 🔴 Sending a value with a different type than `Integer` or `null` for `offset` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+- 🔴 Sending a value with a different type than `Integer` or `null` for `limit` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+- 🔴 Sending a value with a different type than `String` or `null` for `attributesToRetrieve` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
```
Comment (ManyTheFish) [inline text/0124-documents-api.md:118]:
```suggestion
- 🔴 Sending a value with a different type than an array of `String` or `null` for `attributesToRetrieve` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
```
I don't know if you want to precise it.

File: text/0124-documents-api.md [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+
+If the requested `index_uid` does not exist, and the authorization layer allows it (See [3.1.3.4.1. Async Errors](#31341-async-errors)), Meilisearch will create the index when the related asynchronous task resource is executed. See [3.1.3.3. Response Definition](#3133-response-definition).
+
+#### 3.1.4. `PUT` - `indexes/:index_uid/documents`
+
+Add a list of documents or update them if they already exist.
+
+If the provided index does not exist, it will be created. See [3.1.4.5. Lazy Index Creation](#3145-lazy-index-creation)
+
+If an already existing document (same identifier) is set, the old document will be only partially updated according to the fields of the new document. Thus, any fields not present in the new document are kept and remained unchanged.
```
Comment (ManyTheFish) [inline text/0124-documents-api.md:250]:
```suggestion
If an already existing document (same identifier) is set, the old document will be only partially updated according to the fields of the new document. Thus, any fields not present in the new document are kept and remain unchanged.
```

File: text/0124-documents-api.md [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+To completely overwrite a document, see [3.1.3. `POST` - `indexes/:index_uid/documents`](#313-post---indexesindexuiddocuments) section.
```
Comment (ManyTheFish) [inline text/0124-documents-api.md:252]: I don't know if it's relevant to keep this in a spec, for me, it's more a documentation thing.

**Review by ManyTheFish — CHANGES_REQUESTED** [review]

File: text/0124-documents-api.md (hunk 1 above)
Comment (ManyTheFish) [reply text/0124-documents-api.md:118]: but if you want to precise it, I probably miss equivalent error definitions

File: text/0124-documents-api.md (hunk 3 above)
Comment (ManyTheFish) [inline text/0124-documents-api.md:252]: I don't know if it's relevant to keep this in a spec, for me, it's more a documentation thing.

**Comment by gmourier** [comment]
Thanks @ManyTheFish !

**Pull request merged**